### DISCUSSION TITLE
Runnable GlassFish Embedded - run Jakarta EE apps from command line, will be in 7.0.18

### DISCUSSION THREAD

**Discussion by OndroMih** [eclipse-ee4j/glassfish#25159] [Show and tell]
I've recently worked on this new feature in GlassFish. It allows running GlassFish Embedded from command line: `java -jar glassfish-embedded.jar`. This is already merged and will be available in the next version of GlassFish 7.0.18.

Demo:

https://github.com/user-attachments/assets/404d6b30-9271-447a-840a-bea61a89ed7c

Supports configuration via command line arguments and/or config files in the current directory:

* set HTTP port
* deploy applications passed as arguments
* run arbitrary AsAdmin commands at boot
 
All GlassFish Embedded flavors are supported: 

* Standalone glassfish-embedded-all.jar, glassfish-embedded-web.jar, and glassfish-embedded-nucleus.jar
* glassfish-embedded-static-shell.jar in the GlassFish Server installation

Includes fixes that made GlassFish Embedded Web usable. Includes fixes to the Static shell to make it equal to standalone GlassFish Embedded JAR files.

`java -jar glassfish-embedded.jar --help` prints the following info: [glassfish.man.txt](https://github.com/user-attachments/files/17021319/glassfish.man.txt)

The full documentation can be previewed here: https://glassfish.org/docs/SNAPSHOT/embedded-server-guide.html#running-from-command-line.

The pull request with the code changes: https://github.com/eclipse-ee4j/glassfish/pull/25146

**Comment by anbusampath** [comment]
Video missing audio, is it intentional?

**Reply by OndroMih** [reply]
Correct, no audio. It’s just a quick demo.

**Comment by anbusampath** [comment]
What is glassfish-embedded-nucleus.jar? I can understand all and web. is it core profile?

**Reply by OndroMih** [reply]
Good question. The nucleus distribution is something that existed for a long time, at least since GlassFish 3. It has nothing to do with Jakarta EE Core profile. It’s a barebone distribution, I think comparable to a servlet container like Tomcat, and it was designed to allow adding individual Java EE components according to what applications need. However, it’s not documented how to do this.

**Reply by anbusampath** [reply]
> It has nothing to do with Jakarta EE Core profile.

Thanks for the clarification. is there any demand for gf embedded core profile?

> it was designed to allow adding individual Java EE components according to what applications need. However, it’s not documented how to do this.

That would be helpful.
